gensim/utils.py

```diff
@@ -490,6 +497,13 @@ def save(self, fname_or_handle, separately=None, sep_limit=10 * 1024**2,
         in both Python 2 and 3.
 
         """
+        import os
+        directory = os.path.dirname(fname_or_handle)
+        if not os.path.exists(directory):
+            os.makedirs(directory)
+        import corpora.dictionary
+        if isinstance(self, corpora.dictionary.Dictionary):
+            raise TypeError('The method is not built to handle corpora. Use the save_corpus() method')
```

Comment:
The `save` method can handle corpora fine, it's just probably not what the user wants. Plus, you're testing here for `Dictionary` anyway, not corpora.

The recommended way for serializing corpora is `SomeCorpusClass.serialize()`, not `save_corpus()`.